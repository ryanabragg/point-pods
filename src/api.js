import localforage from 'localforage';
import React from 'react';

const api = {
  store: localforage.createInstance({
    name: 'Point Pods',
    storeName: 'point_pods',
    description: 'some description'
  }),
};

export default api;
export const APIContext = React.createContext(api);
export function withAPI(Component) {
  return function APIComponent(props) {
    return (
      <APIContext.Consumer>
        {context => <Component {...props} api={context} />}
      </APIContext.Consumer>
    );
  };
}
