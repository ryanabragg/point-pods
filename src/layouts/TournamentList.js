import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';
import { withAPI } from '../api';

import Divider from '@material-ui/core/Divider';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import DialogActions from '@material-ui/core/DialogActions';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';

import MoreVertIcon from '@material-ui/icons/MoreVert';
import AddIcon from '@material-ui/icons/Add';
import DeleteIcon from '@material-ui/icons/Delete';
import SearchIcon from '@material-ui/icons/Search';
import CloseIcon from '@material-ui/icons/Close';
import SortIcon from '@material-ui/icons/Sort';
import CloudDownloadIcon from '@material-ui/icons/CloudDownload';

import AppMenu from '../components/AppMenu';
import MenuButton from '../components/MenuButton';
import TournamentCards from '../components/TournamentCards';

const styles = theme => ({
  actionButton: {
    margin: 2 * theme.spacing.unit,
    position: 'absolute',
    bottom: 0,
    right: 0,
  },
  dialog: {
    minWidth: theme.breakpoints.values['xs'],
  },
  cancel: {
    color: theme.palette.error.main,
  },
  content: {
    padding: theme.spacing.unit,
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-evenly',
    alignItems: 'flex-start',
    alignContent: 'flex-start',
  },
  right: {
    marginLeft: 'auto',
  },
});

class TournamentList extends Component {
  state = {
    dialog: null,
    newPlayer: '',
    searching: false,
    searchTerm: '',
    category: '',
    status: '',
    sort: {
      field: 'name',
      ascending: true,
    },
    tournaments: [],
    selected: [],
    categories: [],
  };

  componentDidMount() {
    this.props.api.Tournaments.all()
      .then(list => {
        let categories = list.map(tournament => tournament.category)
          .filter((cat, i, self) => cat && self.indexOf(cat) === i).sort();
        this.setState({
          tournaments: list,
          categories: categories,
        });
      })
      .catch(error => this.props.notification(error.message, 'error'));
  }

  sorting = (a, b) => {
    const { field, ascending } = this.state.sort;
    let left = a[field],
      right = b[field];
    if(['name', 'category'].includes(field)) {
      left = left.toString().toLowerCase();
      right = right.toString().toLowerCase();
    }
    if(left < right)
      return ascending ? -1 : 1;
    else if(left > right)
      return ascending ? 1 : -1;
    return 0;
  };

  setSort = (field) => () => {
    let { sort } = this.state;
    if(field === sort.field)
      sort.ascending = !sort.ascending;
    else
      sort = {
        field: field,
        ascending: true,
      };
    this.setState({ sort });
  };

  setDialog = (dialog) => () => {
    this.setState({
      dialog: dialog,
      newPlayer: '',
    });
  };

  setSearching = (bool) => () => {
    let nextState = { searching: bool };
    if(!bool)
      nextState.searchTerm = '';
    this.setState(nextState);
  };

  onInputChange = name => event => {
    const value = event.target.value;
    this.setState({ [name]: value });
  };

  onSelect = (selected) => {
    this.setState({ selected: selected });
  };

  renderToolbar = () => {
    if(this.state.searching) {
      return (
        <React.Fragment>
          <TextField
            autoFocus
            margin='dense'
            id='search'
            label='Search'
            type='text'
            value={this.state.searchTerm}
            onChange={this.onInputChange('searchTerm')}
          />
          <IconButton
            color='inherit'
            onClick={this.setSearching(false)}
          >
            <CloseIcon />
          </IconButton>
        </React.Fragment>
      );
    }
    return (
      <React.Fragment>
        <IconButton
          color={this.state.searchTerm ? 'secondary' : 'inherit'}
          onClick={this.setSearching(true)}
        >
          <SearchIcon />
        </IconButton>
        <MenuButton
          buttonType='icon'
          buttonProps={{color: 'inherit'}}
          buttonContent={<MoreVertIcon />}
          menuItems={[{
            label: 'Sort By Name',
            icon: <SortIcon />,
            action: this.setSort('name'),
          }, {
            label: 'Sort By Date',
            icon: <SortIcon />,
            action: this.setSort('date'),
          }]}
        />
      </React.Fragment>
    );
  };

  renderDialogCategories = () => null;

  render() {
    const { classes } = this.props;
    const { dialog, searching, searchTerm, tournaments, selected } = this.state;
    const cards = Array.from(tournaments).sort(this.sorting);
    let showDialog;
    switch(dialog) {
      case 'categories': showDialog = this.renderDialogCategories(); break;
      default: showDialog = null;
    }
    return (
      <div>
        <AppMenu
          title={searching ? '' : 'Point Pods'}
          color={searching ? 'secondary' : 'primary'}
          toolbar={this.renderToolbar()}
        />
        <Button
          variant='fab'
          color='secondary'
          aria-label='New Tournament'
          className={classes.actionButton}
          href='/new'
        >
          <AddIcon />
        </Button>
        <TournamentCards
          tournaments={cards}
          selected={selected}
          onSelect={this.onSelect}
          search={searchTerm}
        />
        {showDialog}
      </div>
    );
  }
}

TournamentList.defaultProps = {
  notification: (m, v, d, c) => null,
};

TournamentList.propTypes = {
  api: PropTypes.object.isRequired, // added by withAPI
  classes: PropTypes.object.isRequired, // added by withStyles
  notification: PropTypes.func,
};

export default withStyles(styles, { withTheme: true })(withAPI(TournamentList));
