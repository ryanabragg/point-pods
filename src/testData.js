import api from './api';

const testSettings = {
  pairingMethod: api.pairingMethods.HISTORIC,
  pairingMethodInitial: api.pairingMethods.HISTORIC,
  podSizeMinimum: 4,
  podSizeMaximum: 7,
};

const testPlayers = [{
  id: 'mock0',
  name: 'test0',
  points: 0,
}, {
  id: 'mock1',
  name: 'test1',
  points: 1,
}, {
  id: 'mock2',
  name: 'test2',
  points: 2,
}, {
  id: 'mock3',
  name: 'test3',
  points: 3,
}, {
  id: 'mock4',
  name: 'test4',
  points: 4,
}, {
  id: 'mock5',
  name: 'test5',
  points: 5,
}, {
  id: 'mock6',
  name: 'test6',
  points: 6,
}, {
  id: 'mock7',
  name: 'test7',
  points: 7,
}, {
  id: 'mock8',
  name: 'test8',
  points: 8,
}, {
  id: 'mock9',
  name: 'test9',
  points: 9,
}];

const testTournaments = [{
  id: 'mock0',
  name: 'test0',
  category: 'cat0',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vitae tellus nunc. Etiam urna tortor, laoreet vitae commodo porttitor, mattis id purus.',
  date: new Date('2018-01-01'),
  done: false,
  staging: true,
  pairingMethod: api.pairingMethods.RANDOM,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 4,
  rounds: 1,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 3,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 3 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 0,
    participated: false,
    dropped: false,
  }],
}, {
  id: 'mock1',
  name: 'test1',
  category: 'cat1',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed molestie suscipit suscipit. Nullam sollicitudin tempus pretium.',
  date: new Date('2018-02-01'),
  done: true,
  staging: false,
  pairingMethod: api.pairingMethods.POINTS,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 5,
  rounds: 2,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 0,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 0 },
    2: { pod: 1, points: 0 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 1 },
    2: { pod: 2, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
    2: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 3,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 3 },
    2: { pod: 2, points: 3 },
  }, {
    id: 'mock4',
    name: 'test4',
    points: 4,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 4 },
    2: { pod: 1, points: 4 },
  }, {
    id: 'mock5',
    name: 'test5',
    points: 5,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 5 },
    2: { pod: 2, points: 5 },
  }, {
    id: 'mock6',
    name: 'test6',
    points: 6,
    participated: true,
    dropped: false,
    2: { pod: 2, points: 6 },
  }],
}, {
  id: 'mock2',
  name: 'test2',
  category: 'cat0',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vitae tellus nunc. Sed ullamcorper metus vitae nisi rhoncus placerat.',
  date: new Date('2018-01-01'),
  done: false,
  staging: false,
  pairingMethod: api.pairingMethods.RANDOM,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 4,
  rounds: 1,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 0,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 0 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 3,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 3 },
  }],
}];

const mockRoute = {
  history: {
    push: jest.fn(to => to),
    replace: jest.fn(to => to),
  },
  match: {
    params: {},
  },
  matchWithInvalid: {
    params: {
      id: 'invalid',
    },
  },
  matchWithID0: {
    params: {
      id: 'mock0',
    },
  },
  matchWithID1: {
    params: {
      id: 'mock1',
    },
  },
};

let mockStore = {};
const mockStorage = {
  setItem: jest.fn((key, val) => Promise.resolve((mockStore[key] = val))),
  getItem: jest.fn(key => Promise.resolve(mockStore[key])),
  removeItem: jest.fn(key => { delete mockStore[key]; Promise.resolve(); }),
  clear: jest.fn(() => Promise.resolve((mockStore = {}))),
  keys: jest.fn(() => Promise.resolve(Object.keys(mockStore))),
  length: jest.fn(() => Promise.resolve(Object.keys(mockStore).length)),
};

let defaultStore = null;
function mockAPI(api, storage) {
  defaultStore = api.store;
  api.store = storage;
  mockStore = {
    settings: testSettings,
    players: testPlayers,
    tournaments: testTournaments,
  };
}
function demockAPI(api) {
  api.store = defaultStore;
  defaultStore = null;
}

export {
  testSettings,
  testPlayers,
  testTournaments,
  mockRoute,
  mockStore,
  mockStorage,
  mockAPI,
  demockAPI,
};
