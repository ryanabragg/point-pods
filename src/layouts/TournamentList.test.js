import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Dialog from '@material-ui/core/Dialog';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';

import TournamentList from './TournamentList';

import api from '../api';

import { testTournaments, mockStorage, mockAPI, demockAPI } from '../testData';

beforeEach(() => mockAPI(api, mockStorage));
afterEach(() => demockAPI(api));

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TournamentList />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('matches the prior snapshot', (done) => {
  const component = renderer.create(<TournamentList />);
  expect(component.toJSON()).toMatchSnapshot();
  component.update(<TournamentList />);
  setTimeout(() => {
    try {
      expect(component.toJSON()).toMatchSnapshot();
      done();
    } catch (error) {
      done.fail(error);
    }
  }, 100);
});

test('is a HOC-ed component', () => {
  const wrapper = shallow(<TournamentList />);
  expect(wrapper.type().prototype.constructor.name).toBe('APIComponent');
  expect(wrapper.dive().type().prototype.constructor.name).toBe('TournamentList');
});

describe('defaults', () => {
  test('values', () => {
    const wrapper = shallow(<TournamentList />);
    const component = wrapper.dive().dive();
    expect(component.state('dialog')).toBe(null);
    expect(component.state('searching')).toBe(false);
    expect(component.state('sort')).toEqual({field: 'name', ascending: true});
    expect(component.state('selected')).toEqual([]);
  });

  test('loads the state', (done) => {
    const wrapper = shallow(<TournamentList />);
    const component = wrapper.dive().dive();
    expect(component.state('tournaments')).toEqual([]);
    expect(component.state('categories')).toEqual([]);
    setTimeout(() => {
      try {
        expect(component.state('tournaments')).toEqual(testTournaments);
        expect(component.state('categories')).toEqual(testTournaments.map(t => t.category)
          .filter((cat, i, self) => cat && self.indexOf(cat) === i).sort());
        done();
      } catch (error) {
        done.fail(error);
      }
    }, 100);
  });
});

describe('actions', () => {
  test('select');

  test('searching');

  test('filter by category');

  test('filter by status');

  test('sorting');

  test('add a tournament', () => {
    const wrapper = shallow(<TournamentList />);
    const component = wrapper.dive().dive();
    expect(component.findWhere(n => n.prop('variant') === 'fab').prop('href')).toBe('/new');
  });
});
