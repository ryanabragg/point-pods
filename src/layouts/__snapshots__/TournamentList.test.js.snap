// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the prior snapshot 1`] = `
<div>
  <div
    className="AppMenu-root-153"
  >
    <header
      className="MuiPaper-root-168 MuiPaper-elevation4-174 MuiAppBar-root-160 MuiAppBar-positionFixed-161 MuiAppBar-colorPrimary-166 mui-fixed"
    >
      <div
        className="MuiToolbar-root-195 MuiToolbar-regular-197 MuiToolbar-gutters-196 AppMenu-nowrap-155"
      >
        <button
          aria-label="Menu"
          className="MuiButtonBase-root-205 MuiIconButton-root-199 MuiIconButton-colorInherit-200 AppMenu-menuButton-156"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-204"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-208"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-215"
          />
        </button>
        <h2
          className="MuiTypography-root-222 MuiTypography-title-228 MuiTypography-colorInherit-241 AppMenu-flex-154"
        >
          Point Pods
        </h2>
        <button
          className="MuiButtonBase-root-205 MuiIconButton-root-199 MuiIconButton-colorInherit-200"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-204"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-208"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-215"
          />
        </button>
        <div>
          <button
            aria-haspopup="true"
            aria-owns={null}
            className="MuiButtonBase-root-205 MuiIconButton-root-199 MuiIconButton-colorInherit-200"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-204"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-208"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-215"
            />
          </button>
        </div>
      </div>
    </header>
    <div
      className="AppMenu-spacer-158"
    />
  </div>
  <a
    aria-label="New Tournament"
    className="MuiButtonBase-root-205 MuiButton-root-261 MuiButton-fab-276 MuiButton-contained-270 MuiButton-containedSecondary-272 MuiButton-raised-273 MuiButton-raisedSecondary-275 APIComponent-actionButton-148"
    href="/new"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiButton-label-262"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-208"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-215"
    />
  </a>
  <div
    className="TournamentCards-root-285"
  >
    <div
      className="MuiPaper-root-168 MuiPaper-elevation1-171 MuiPaper-rounded-169 MuiCard-root-288 TournamentCards-card-286"
      id="not-found"
    >
      <div
        className="MuiCardHeader-root-289"
      >
        <div
          className="MuiCardHeader-content-292"
        >
          <span
            className="MuiTypography-root-222 MuiTypography-headline-227 MuiCardHeader-title-293"
          >
            No Tournaments Found
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`matches the prior snapshot 2`] = `
<div>
  <div
    className="AppMenu-root-153"
  >
    <header
      className="MuiPaper-root-168 MuiPaper-elevation4-174 MuiAppBar-root-160 MuiAppBar-positionFixed-161 MuiAppBar-colorPrimary-166 mui-fixed"
    >
      <div
        className="MuiToolbar-root-195 MuiToolbar-regular-197 MuiToolbar-gutters-196 AppMenu-nowrap-155"
      >
        <button
          aria-label="Menu"
          className="MuiButtonBase-root-205 MuiIconButton-root-199 MuiIconButton-colorInherit-200 AppMenu-menuButton-156"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-204"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-208"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-215"
          />
        </button>
        <h2
          className="MuiTypography-root-222 MuiTypography-title-228 MuiTypography-colorInherit-241 AppMenu-flex-154"
        >
          Point Pods
        </h2>
        <button
          className="MuiButtonBase-root-205 MuiIconButton-root-199 MuiIconButton-colorInherit-200"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-204"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-208"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-215"
          />
        </button>
        <div>
          <button
            aria-haspopup="true"
            aria-owns={null}
            className="MuiButtonBase-root-205 MuiIconButton-root-199 MuiIconButton-colorInherit-200"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-204"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-208"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-215"
            />
          </button>
        </div>
      </div>
    </header>
    <div
      className="AppMenu-spacer-158"
    />
  </div>
  <a
    aria-label="New Tournament"
    className="MuiButtonBase-root-205 MuiButton-root-261 MuiButton-fab-276 MuiButton-contained-270 MuiButton-containedSecondary-272 MuiButton-raised-273 MuiButton-raisedSecondary-275 APIComponent-actionButton-148"
    href="/new"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiButton-label-262"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-208"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-215"
    />
  </a>
  <div
    className="TournamentCards-root-285"
  >
    <div
      className="MuiPaper-root-168 MuiPaper-elevation1-171 MuiPaper-rounded-169 MuiCard-root-288 TournamentCards-card-286"
      id="mock0"
    >
      <div
        className="MuiCardHeader-root-289"
      >
        <div
          className="MuiCardHeader-content-292"
        >
          <span
            className="MuiTypography-root-222 MuiTypography-headline-227 MuiCardHeader-title-293"
          >
            test0
          </span>
          <span
            className="MuiTypography-root-222 MuiTypography-body1-231 MuiTypography-colorTextSecondary-245 MuiCardHeader-subheader-294"
          >
            cat0
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root-295"
      >
        <p
          className="MuiTypography-root-222 MuiTypography-body1-231"
        >
          4 Total Players
        </p>
        <p
          className="MuiTypography-root-222 MuiTypography-body1-231"
        >
          3 Active Players
        </p>
      </div>
      <div
        className="MuiCardActions-root-296"
      >
        <a
          className="MuiButtonBase-root-205 MuiButton-root-261 MuiButton-text-263 MuiButton-flat-266 MuiButton-sizeSmall-282 TournamentCards-right-287 MuiCardActions-action-297"
          href="/tournament/mock0"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-262"
          >
            Continue
          </span>
          <span
            className="MuiTouchRipple-root-215"
          />
        </a>
      </div>
    </div>
    <div
      className="MuiPaper-root-168 MuiPaper-elevation1-171 MuiPaper-rounded-169 MuiCard-root-288 TournamentCards-card-286"
      id="mock1"
    >
      <div
        className="MuiCardHeader-root-289"
      >
        <div
          className="MuiCardHeader-content-292"
        >
          <span
            className="MuiTypography-root-222 MuiTypography-headline-227 MuiCardHeader-title-293"
          >
            test1
          </span>
          <span
            className="MuiTypography-root-222 MuiTypography-body1-231 MuiTypography-colorTextSecondary-245 MuiCardHeader-subheader-294"
          >
            cat1
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root-295"
      >
        <p
          className="MuiTypography-root-222 MuiTypography-body1-231"
        >
          8 Total Players
        </p>
        <p
          className="MuiTypography-root-222 MuiTypography-body1-231"
        >
          6 Active Players
        </p>
      </div>
      <div
        className="MuiCardActions-root-296"
      >
        <a
          className="MuiButtonBase-root-205 MuiButton-root-261 MuiButton-text-263 MuiButton-flat-266 MuiButton-sizeSmall-282 TournamentCards-right-287 MuiCardActions-action-297"
          href="/tournament/mock1"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-262"
          >
            Details
          </span>
          <span
            className="MuiTouchRipple-root-215"
          />
        </a>
      </div>
    </div>
    <div
      className="MuiPaper-root-168 MuiPaper-elevation1-171 MuiPaper-rounded-169 MuiCard-root-288 TournamentCards-card-286"
      id="mock2"
    >
      <div
        className="MuiCardHeader-root-289"
      >
        <div
          className="MuiCardHeader-content-292"
        >
          <span
            className="MuiTypography-root-222 MuiTypography-headline-227 MuiCardHeader-title-293"
          >
            test2
          </span>
          <span
            className="MuiTypography-root-222 MuiTypography-body1-231 MuiTypography-colorTextSecondary-245 MuiCardHeader-subheader-294"
          >
            cat0
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root-295"
      >
        <p
          className="MuiTypography-root-222 MuiTypography-body1-231"
        >
          4 Total Players
        </p>
        <p
          className="MuiTypography-root-222 MuiTypography-body1-231"
        >
          4 Active Players
        </p>
      </div>
      <div
        className="MuiCardActions-root-296"
      >
        <a
          className="MuiButtonBase-root-205 MuiButton-root-261 MuiButton-text-263 MuiButton-flat-266 MuiButton-sizeSmall-282 TournamentCards-right-287 MuiCardActions-action-297"
          href="/tournament/mock2"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-262"
          >
            Continue
          </span>
          <span
            className="MuiTouchRipple-root-215"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
